src/Extensions/Logging/Internal/Replayable/ReplayableActivityProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.Omex.Extensions.TimedScopes;
+
+namespace Microsoft.Omex.Extensions.Logging.Replayable
+{
+	/// <summary>
+	/// Depending on <see cref="OmexLoggingOptions" /> class will create regular <see cref="Activity"/>
+	/// or <see cref="ReplayableActivity" /> that is able to aggregate log messages for replay
+	/// </summary>
+	internal class ReplayableActivityProvider : IActivityProvider
+	{
+		public ReplayableActivityProvider(IOptions<OmexLoggingOptions> options)
+		{
+			m_replayLogsInCaseOfError = options.Value.ReplayLogsInCaseOfError;
+			m_maxReplayedEventsPerActivity = options.Value.MaxReplayedEventsPerActivity;
+		}
+
```

Comment:
Nitpick: add another blank line for consistency

Reply:
I will resolve it in the follow-up pr https://github.com/microsoft/Omex/pull/168